tests/test_basic_auth.py

```diff
@@ -0,0 +1,143 @@
+    """
+    version_info = {
+        "buildId": "12345",
+        "buildUrl": "https://some.server.com/domino/12345",
+        "commitId": "123deadbeef456",
+        "commitUrl": "https://some.repo.com/domino/commit/123deadbeef456",
+        "timestamp": "2021-06-14T16:50:02Z",
+        "version": "9.9.9"
+    }
+
+    with requests_mock.Mocker() as mock_endpoint:
+        mock_endpoint.get("http://mock.domino.com/version", json=version_info, status_code=200)
+        yield
+
+
+@pytest.fixture
+def dummy_token_file(tmpdir):
+    """
+    Simulate a token file.
+    """
+    token_file = tmpdir.join("dummy_token_file.txt")
+    token_file.write('top_secret_auth_token')
+    return token_file
+
+
+@pytest.fixture
+def real_token_file(tmpdir):
+    """
+    Derive a real authentication token (assumes that certain packages are available).
+    If any exception is thrown, returning None can signal tests to XFAIL.
+    """
+    try:
+        from domino_pytk.authentication import KeycloakToken
+        _url = os.getenv('DOMINO_HOST_KEY_NAME')
+        _user = os.getenv('DOMINO_USER')
+        _pw = os.getenv('DOMINO_PASSWORD')
+        keycloak_token = KeycloakToken(url=_url, username=_user, password=_pw)
+
+        token_file = tmpdir.join("tmp_token_file.txt")
+        token_file.write(keycloak_token)
+        os.environ['DOMINO_TOKEN_FILE_KEY_NAME'] = str(token_file)
+        return str(token_file)
+    except Exception as e:
+        print(e)
+        return None
+
+
+@pytest.mark.usefixtures("mock_domino_version_response")
+def test_object_creation_with_api_key():
+    """
+    Confirm that the expected auth type is used when using an api key.
+    """
+    dummy_host = "http://mock.domino.com"
+    dummy_user_project = "user_name/project_name"
+    dummy_api_key = "top_secret_api_key"
+
+    d = Domino(host=dummy_host, project=dummy_user_project, api_key=dummy_api_key)
+    assert isinstance(d.request_manager.auth, requests.auth.HTTPBasicAuth)
+
+
+@pytest.mark.usefixtures("mock_domino_version_response")
+def test_object_creation_with_token_file(dummy_token_file):
+    """
+    Confirm that the expected auth type is used when using a token file.
+    """
+    dummy_host = "http://mock.domino.com"
+    dummy_user_project = "user_name/project_name"
+
+    d = Domino(host=dummy_host, project=dummy_user_project, domino_token_file=dummy_token_file)
+    assert isinstance(d.request_manager.auth, bearer_auth.BearerAuth)
+
+
+@pytest.mark.skipif(not domino_is_reachable(), reason="No access to a live Domino deployment")
+@pytest.mark.skipif(not os.getenv('DOMINO_API_KEY_KEY_NAME'), reason="No API key in environment")
+def test_auth_against_real_deployment_with_api_key():
+    '''
+    Confirm against a live system that validating by API key works.
+
+    Assumes that ${DOMINO_HOST_KEY_NAME} contains a valid Domino URL
+    Assumes that ${DOMINO_API_KEY_KEY_NAME} contains a valid API key
```

Comment:
Here and elsewhere, can the environment variable names align with the rest of the repo?
```suggestion
    Assumes that ${DOMINO_API_HOST} contains a valid Domino URL
    Assumes that ${DOMINO_USER_API_KEY} contains a valid API key
```